corehq/motech/generic_inbound/reports.py

```diff
@@ -0,0 +1,100 @@
+from django.utils.functional import cached_property
+from django.utils.translation import gettext as _
+from django.utils.translation import gettext_lazy
+
+from corehq.apps.reports.datatables import DataTablesColumn, DataTablesHeader
+from corehq.apps.reports.dispatcher import DomainReportDispatcher
+from corehq.apps.reports.filters.base import BaseMultipleOptionFilter, BaseSimpleFilter
+from corehq.apps.reports.generic import GenericTabularReport
+from corehq.apps.reports.standard import DatespanMixin
+from corehq.toggles import GENERIC_INBOUND_API
+
+from .models import RequestLog, ProcessingAttempt
+
+
+class RequestStatusFilter(BaseMultipleOptionFilter):
+    slug = 'request_status'
+    label = gettext_lazy('Request Status')
+
+    @property
+    def options(self):
+        return RequestLog.Status.choices
+
+
+class FormIdFilter(BaseSimpleFilter):
+    slug = "form_id"
+    label = gettext_lazy("Form ID")
+    help_inline = gettext_lazy("Enter a form id to filter results")
+
+
+class ApiRequestLogReport(DatespanMixin, GenericTabularReport):
+    name = gettext_lazy('Inbound API Request Logs')
+    slug = 'api_request_log_report'
+    base_template = "reports/base_template.html"
+    section_name = gettext_lazy('Project Settings')
+    dispatcher = DomainReportDispatcher
+    ajax_pagination = True
+    sortable = False
+
+    fields = [
+        'corehq.apps.reports.filters.dates.DatespanFilter',
+        'corehq.motech.generic_inbound.reports.RequestStatusFilter',
+        'corehq.motech.generic_inbound.reports.FormIdFilter',
+    ]
+
+    toggles = [GENERIC_INBOUND_API]
+
+    @classmethod
+    def allow_access(cls, request):
+        return request.couch_user.is_domain_admin()
+
+    @property
+    def total_records(self):
+        return self._queryset.count()
+
+    @property
+    def shared_pagination_GET_params(self):
+        return [
+            {'name': param, 'value': self.request.GET.getlist(param)}
+            for param in ['request_status', 'startdate', 'enddate', 'form_id']
+        ]
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(_("API")),
+            DataTablesColumn(_("Timestamp")),
+            DataTablesColumn(_("Status")),
+            DataTablesColumn(_("Response")),
+            DataTablesColumn(_("Form ID")),
+        )
+
+    @cached_property
+    def _queryset(self):
+        queryset = RequestLog.objects.filter(
+            domain=self.domain,
+            timestamp__gte=self.request.datespan.startdate_utc,
+            timestamp__lt=self.request.datespan.enddate_utc,
```

Comment:
Do we want this to be inclusive of the last date? Peaking around the code that's usually what is usually done for end date and in my mind is the most intuitive. If I say I want data from Mon-Fri, I want to see the data from Friday too

Reply:
IMO it'd be nice to rename `datespan.enddate` to `datespan._enddate_raw` and never use it outside of the class.

Reply:
@stephherbers Since this was blocking I just wanted to double check before resolving that we are good with the `datespan.endate` functionality here, that it is actually inclusive.